modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java

```diff
@@ -548,6 +540,29 @@ private List<ObjectValidationRule> _getObjectValidationRules(
 		return objectValidationRules;
 	}
 
+	private List<ObjectValidationRule> _getObjectValidationRulesDSLQuery(
+		boolean active, long companyId, List<String> externalReferenceCodes,
+		long objectDefinitionId) {
+
+		DSLQuery dslQuery = DSLQueryFactoryUtil.select(
+		).from(
+			ObjectValidationRuleTable.INSTANCE
+		).where(
+			ObjectValidationRuleTable.INSTANCE.objectDefinitionId.eq(
+				objectDefinitionId
+			).and(
+				ObjectValidationRuleTable.INSTANCE.active.eq(active)
+			).and(
+				ObjectValidationRuleTable.INSTANCE.companyId.eq(companyId)
+			).and(
+				ObjectValidationRuleTable.INSTANCE.externalReferenceCode.in(
+					externalReferenceCodes.toArray(new String[0]))
```

Comment:
@JhossephG please follow the same order from [here](https://github.com/liferay/liferay-portal/blob/0f991ed1a929917f20a655dc130ad21c46a5a891/modules/apps/object/object-service/src/main/resources/META-INF/sql/tables.sql#L365-L373)